### PR TITLE
initial pr

### PR DESCRIPTION
**Pull request by starikcetin**

### CODE REVIEW

File: src/components/pwc-choices-2/pwc-choices-2.tsx [hunk 1]
```diff
@@ -1,32 +1,192 @@
...
+    mode: PwcChoices2.RetreiveMode = "option"
+  ): Promise<string[] | PwcChoices2.IOption[]> {
+    switch (mode) {
+      case "option":
+        return this.selectedOptions;
+
+      case "value":
+        return this.selectedOptions.map(o => o.value);
+
+      case "option":
```
Comment (kuraydev) [inline src/components/pwc-choices-2/pwc-choices-2.tsx]: Fix `getSelectedOptions`

File: src/components/pwc-choices-2/pwc-choices-2.tsx [hunk 2]
```diff
@@ -1,32 +1,192 @@
...
-    return <div>Hello, World! I'm {this.getText()}</div>;
+    return (
+      <div class="container">
+        {this.constructInputBar()}
+        {this.dropdownIsOpen && this.constructDropdown()}
+      </div>
+    );
+  }
+
+  constructInputBar() {
```
Comment (kuraydev) [inline src/components/pwc-choices-2/pwc-choices-2.tsx:174]: `constructInputBar` and `constructDropdown` should be before the render

File: src/interfaces/PwcChoices2.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+export namespace PwcChoices2 {
+  export const AllDistinctModeLiterals = [
```
Comment (kuraydev) [inline src/interfaces/PwcChoices2.ts:2]: `getOptionsSwitchMethod` these enums not included in the library.

File: src/components/pwc-choices-2-option-bubble/pwc-choices-2-option-bubble.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    return (
+      <div class="bubble">
+        <span class="bubble-label">{this.option.label}</span>
+        {this.showCloseButton && (
+          <span
+            class="bubble-closeButton"
+            onClick={e => this.onCloseClicked(e)}
+          >
+            <svg
+              xmlns="http://www.w3.org/2000/svg"
```
Comment (kuraydev) [inline src/components/pwc-choices-2-option-bubble/pwc-choices-2-option-bubble.tsx]: Could be removed.

File: src/components/pwc-choices-2-dropdown/pwc-choices-2-dropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+     * but we have to.
+     */
+    this.options.splice(this.options.indexOf(originalOption), 1);
+    this.filteredOptions.splice(
+      this.filteredOptions.indexOf(optionFilterResult),
+      1
+    );
+  }
+
+  makeOptionsWholeFilterResult() {
```
Comment (kuraydev) [inline src/components/pwc-choices-2-dropdown/pwc-choices-2-dropdown.tsx]: rename the function `convertAllOptionsToFilterResult`

**Review by kuraydev — APPROVED** [review]

File: src/components/pwc-choices-2-dropdown/pwc-choices-2-dropdown.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+      </div>
+    );
+  }
+
+  onSearchInput(e: Event): void {
+    e.stopPropagation();
+    const wholeSearchInput = this.getWholeSearchInput();
+
+    if (wholeSearchInput.length === 0) {
+      this.filteredOptions = this.makeOptionsWholeFilterResult();
```
Comment (kuraydev) [inline src/components/pwc-choices-2-dropdown/pwc-choices-2-dropdown.tsx]: `makeOptionsWholeFilterResult` change it all

**Review by kuraydev — CHANGES_REQUESTED** [review]

File: src/components/pwc-choices-2/pwc-choices-2.tsx (hunk 1 above)
Comment (starikcetin) [reply src/components/pwc-choices-2/pwc-choices-2.tsx]: fixed: cb82c87966931811b832907ebe9da0e2cce1508f

File: src/components/pwc-choices-2/pwc-choices-2.tsx (hunk 2 above)
Comment (starikcetin) [reply src/components/pwc-choices-2/pwc-choices-2.tsx:174]: fixed: c08b9a9d6c5ab24613cd90835a75fb7508d8fcd3

File: src/components/pwc-choices-2-option-bubble/pwc-choices-2-option-bubble.tsx (hunk 4 above)
Comment (starikcetin) [reply src/components/pwc-choices-2-option-bubble/pwc-choices-2-option-bubble.tsx]: fixed: ecb3acd67e76d92007a2759777423297891d240d

File: src/components/pwc-choices-2-dropdown/pwc-choices-2-dropdown.tsx (hunk 5 above)
Comment (starikcetin) [reply src/components/pwc-choices-2-dropdown/pwc-choices-2-dropdown.tsx]: a17975ce7d50a96f9eaf78d26f7c23366f4f9d5e

File: src/components/pwc-choices-2-dropdown/pwc-choices-2-dropdown.tsx (hunk 6 above)
Comment (starikcetin) [reply src/components/pwc-choices-2-dropdown/pwc-choices-2-dropdown.tsx]: a17975ce7d50a96f9eaf78d26f7c23366f4f9d5e

**Pull request merged**